core/src/main/java/org/axonframework/deadline/quartz/DeadlineJob.java

```diff
@@ -101,7 +106,7 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
             }
         } catch (Exception e) {
```

Comment:
Ah bummer, I notice this retry is now done on _any_ `Exception`.
This is not desirable however, as there are plenty of exception for which a retry doesn't make sense.

It would be better if you'd use the `ExceptionUtils#findException` function, providing `e` and a `Predicate` that checks whether the exception or one of it's causes is `AxonNonTransientException`. Only if the returned `Optional` is empty is it allowed to use the retry, as that means you have a transient exception, like the `ConcurrencyException` your original user group message was about. 

It would be beneficial if you could provide a test case for this scenario as well by the way.

Reply:
Thanks for the adjustment @aupodogov :+1: